opnsense_api/util/controller.py

```diff
@@ -1,24 +1,24 @@
-import urllib.parse
 import logging
+import urllib.parse
 
 log = logging.getLogger(__name__)
 
 
-class OPNsenseAPIController:
-    def __init__(self, device, module, controller):
+class OPNSenseAPIController:
+    def __init__(self, device, module: str, controller: str):
         self._device = device
         self._module = module
         self._controller = controller
 
-    def _format_request(self, command: str, uuid: str = None, params=None) -> str:
+    def _format_request(self, command: str, resource_identifier: str = None, params: dict = None) -> str:
         # Simplest url path for a request
         # e.g. api/unbound/settings/searchHostOverride
         base_request = f"{self._module}/{self._controller}/{command}"
 
         # Add in the UUID for a specific resource
         # e.g. api/unbound/settings/getHostOverride/2ce9672e-43a5-4462-9cf1-084964970862
-        if uuid is not None:
-            base_request = f"{self._module}/{self._controller}/{command}/{uuid}"
+        if resource_identifier is not None:
+            base_request = f"{base_request}/{resource_identifier}"
```

Comment:
What was the motivation behind this? Is this change related to the edge cases you mentioned?

Reply:
Well for one I wanted to not recreate what we already have in base_request. But I guess your question is related to uuid -> resource_identifier. I think that came from me mistyping a uuid during API testing and thinking: "Oh, it seems like it isn't always the uuid. I'll look over everything again, and can revert it if it is indeed the uuid everywhere, if you wish.

Reply:
lets revert that, resource_identifier is going to be confusing in the long run.